apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -692,12 +698,32 @@ export class RushConfiguration {
     );
     this._versionPolicyConfiguration = new VersionPolicyConfiguration(versionPolicyConfigFile);
 
+    const variants: {
+      [variantName: string]: boolean;
+    } = {};
+
+    if (rushConfigurationJson.variants) {
+      for (const variantOptions of rushConfigurationJson.variants) {
+        const { variantName } = variantOptions;
+
+        if (variants[variantName]) {
+          throw new Error(`Duplicate variant named '${variantName}' specified in configuration.`);
+        }
+
+        variants[variantName] = true;
+      }
+    }
+
+    this._variants = variants;
+  }
```

Comment:
Would this be faster if you used a `Set<string>`?

Reply:
`variants` is supposed to be a really really small list, yes? The perf diff might be negligible.

Reply:
@sachinjoseph  `Set<string>` and `Map<string, boolean>` both also the advantage of being correct code. :-)

Your code here will malfunction if someone's Rush variant has a name like `constructor` that accidentally conflicts with a member of the `Object` prototype.

Reply:
I can change this to use a `Set<T>`.